File: src/#1512/solution.js
var numIdenticalPairs = function (nums) {
    let numMap = new Map()

    nums.forEach((num) => {
        if (numMap.has(num)) {
            numMap.set(num, numMap.get(num) + 1)
        } else {
            numMap.set(num, 0)
        }
    })
    let res = 0
    numMap.forEach((count) => {
        res += ((1 + count) * count) / 2
    })
    return res
}

File: src/#1512/solution.go
package solution

func numIdenticalPairs(nums []int) int {
	numMap := make(map[int]int)

	for _, num := range nums {
		if _, ok := numMap[num]; ok {
			numMap[num] += 1
		} else {
			numMap[num] = 0
		}
	}
	res := 0

	for _, count := range numMap {
		res += ((1 + count) * count) / 2
	}

	return res
}
